packages/mcp/src/registry/config-generator.ts

```diff
@@ -0,0 +1,346 @@
+          // Default behavior
+          entry.command = 'docker';
+          entry.args = [
+            'run',
+            '-i',
+            '--rm',
+            pkg.identifier,
+            ...(pkg.package_arguments || []),
+          ];
+        }
+        handled = true;
+        break;
+      case 'github':
+        if (pkg.runtime_hint) {
+          entry.command = pkg.runtime_hint;
+          entry.args = [
+            ...(pkg.runtime_arguments || []),
+            `github:${pkg.identifier}`,
+            ...(pkg.package_arguments || []),
+          ];
+        } else {
+          // Default behavior
+          entry.command = 'npx';
+          entry.args = [
+            '-y',
+            `github:${pkg.identifier}`,
+            ...(pkg.package_arguments || []),
+          ];
+        }
+        handled = true;
+        break;
+    }
+
+    if (handled) {
+      // Convert environment variables from array to object format
+      // Only include variables that have values (exclude required-only vars)
+      if (pkg.environment_variables && pkg.environment_variables.length > 0) {
+        entry.env = {};
+        for (const envVar of pkg.environment_variables) {
+          if (envVar.value !== undefined) {
+            entry.env[envVar.name] = envVar.value;
+          }
+        }
+      }
+      return entry;
+    }
+  }
+
+  // Handle remote servers
+  if (server.remotes && server.remotes.length > 0) {
+    const remote = server.remotes[0];
+    entry.transport = remote.type;
+    entry.url = remote.url;
+    if (remote.headers && remote.headers.length > 0) {
+      entry.headers = remote.headers;
+    }
+    return entry;
+  }
+
+  // Fallback for unknown types or missing configuration
+  // Use structured clone to create a deep copy and ensure type safety
+  entry._raw_metadata = JSON.parse(JSON.stringify(server));
+  return entry;
+}
+
+/**
+ * Helper to generate command and args JSON lines for install instructions
+ * Eliminates redundancy across npm, pypi, and oci package types
+ */
+function generateCommandArgsLines(
+  pkg: Package,
+  defaultCommand: string,
+  defaultArgs: string[],
+): { lines: string[]; hasEnvVars: boolean } {
+  const lines: string[] = [];
+  const command = pkg.runtime_hint || defaultCommand;
+  lines.push(`  "command": "${command}",`);
+
+  let argsArray: string[];
+  if (pkg.runtime_hint) {
+    // Publisher has full control when hint is provided
+    argsArray = [
+      ...(pkg.runtime_arguments || []),
+      pkg.identifier,
+      ...(pkg.package_arguments || []),
+    ];
+  } else {
+    // Default behavior when no hint provided
+    argsArray = [...defaultArgs, ...(pkg.package_arguments || [])];
+  }
+
+  // Check if environment variables follow
+  const hasEnvVars =
+    pkg.environment_variables &&
+    pkg.environment_variables.some((env) => env.value !== undefined);
+  const argsComma = hasEnvVars ? ',' : '';
+  lines.push(`  "args": ["${argsArray.join('", "')}"]${argsComma}`);
```

Comment:
`generateInstallInstructions` is assembling the JSON snippet with raw string interpolation. On this line we emit `"args": ["${argsArray.join('\", \"')}"]`. If any runtime or package argument contains a double quote, the generated snippet becomes invalid JSON. For example, a registry package that needs an argument like `--flag="value"` would output `"args": ["--flag="value""]`, which breaks parsing when the user follows the instructions. The same issue exists a few lines below for remote headers (line 206) where header values are inserted without escaping. Could we build these arrays/objects with `JSON.stringify` (per element) instead of manual string joining so that quoted arguments/headers produce valid JSON?